teuthology/task/install/rpm.py

```diff
@@ -93,19 +109,26 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
     :param rpm: list of packages names to install
     :param config: the config dict
     """
-    rpm = _package_overrides(rpm, remote.os)
-    builder = _get_builder_project(ctx, remote, config)
-    log.info('Pulling from %s', builder.base_url)
-    log.info('Package version is %s', builder.version)
+    remote_os = remote.os
+    dist_release = remote_os.name
+    rpm = _package_overrides(rpm, remote_os)
     log.info("Installing packages: {pkglist} on remote rpm {arch}".format(
-        pkglist=", ".join(rpm), arch=builder.arch))
-    builder.install_repo()
+        pkglist=", ".join(rpm), arch=remote.arch))
 
-    dist_release = builder.dist_release
-    project = builder.project
+    if 'repos' in config:
+        repos = config.get('repos')
+        log.debug("Adding repos: %s" % repos)
+        if dist_release in ['opensuse', 'sle']:
+            _zypper_addrepo(remote, repos)
```

Comment:
Would it make sense to add an `else:` clause that would raise an exception - something like "Custom repos were specified for OS $remote_os but custom repos are currently supported only on openSUSE and SLE"

Reply:
Sure, good point. Which kind of exception is it preferable here?

Reply:
I have no preference. How about simply `raise Exception( . . . )` ? The idea is to halt execution.